test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -162,43 +205,25 @@ def test_multi_mom_hook_failure_affects_vnode(self):
             NODE, {ATTR_NODE_state: 'offline'},
             id=self.moms.values()[0].shortname)
         self.server.expect(NODE, {ATTR_NODE_state: 'free'},
-                           id=self.moms.values()[1].shortname)
+                           id=self.moms.values()[1].shortname, max_attempts=3,
+                           interval=2)
 
-        vname = "vnode[0]"
         if self.is_cray is True:
             vlist = []
-            first_vnode = 1
             vnl = self.server.filter(
                 VNODE, {'resources_available.vntype': 'cray_compute'})
             vlist = vnl["resources_available.vntype=cray_compute"]
             # Loop through each compute vnode in the list and check if state =
             # free
             for v1 in vlist:
-                if first_vnode == 1:
-                    first_vnode += 1
-                    vname = v1
                 # Check that the node is in free state
                 self.server.expect(
                     VNODE, {'state': 'free'}, id=v1, max_attempts=3,
                     interval=2)
 
         else:
-            self.server.expect(NODE, {ATTR_NODE_state: 'free'}, id='vnode[0]')
-            self.server.expect(NODE, {ATTR_NODE_state: 'free'}, id='vnode[1]')
-
-        mom_host = self.moms.values()[0].shortname
-        pbs_exec = self.server.pbs_conf['PBS_EXEC']
-        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
-        pbsnodes_reset = pbsnodes_cmd + ' -r ' + mom_host
-
-        self.server.delete(jid, wait=True)
-        self.server.manager(MGR_CMD_SET, HOOK, {'enabled': 'False'}, id="h1")
-
-        j2 = Job(TEST_USER)
-        j2.set_attributes({ATTR_l + '.select': '1:vnode=' + vname})
-        jid2 = self.server.submit(j2)
-        self.du.run_cmd(self.server.hostname, cmd=pbsnodes_reset, sudo=True)
-        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
-                           id=self.moms.values()[0].shortname)
-        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'})
-        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid2)
+            self.server.expect(NODE, {ATTR_NODE_state: 'free'}, id='vnode[0]',
+                               max_attempts=3, interval=2)
+            self.server.expect(NODE, {ATTR_NODE_state: 'free'}, id='vnode[1]',
+                               max_attempts=3, interval=2)
+        self.server.manager(MGR_CMD_DELETE, NODE, id="@default")

```

Comment:
Out of curiosity, why do we have to delete the nodes?  Won't the next setUp do that anyway?

Reply:
You are right, this is not needed.